### PR TITLE
Kubectl convert - warn users with NotRegisteredError and Fail on all other errors

### LINKED ISSUE #1383: Kubectl convert (pluggin) doesn't correctly handle manifests with multiple kinds
**What happened**:
running the command
`kubectl convert -f manifest.yaml --output-version 'networking.k8s.io/v1'`

against a file with this content
```yaml
apiVersion: v1
kind: Service
metadata:
  name: uat-odata
  namespace: uat-front
spec:
  clusterIP: None
  clusterIPs:
  - None
  ports:
  - name: port
    port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: odata
    env: uat
    instance: front
  sessionAffinity: None
  type: ClusterIP

---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: uat-ingress
  namespace: uat-front
spec:
  rules:
  - host: uat.company.com
    http:
      paths:
      - backend:
          serviceName: uat-odata
          servicePort: 80
        path: /odata
        pathType: ImplementationSpecific
      - backend:
          serviceName: webapi
          servicePort: 80
        path: /api
        pathType: ImplementationSpecific
```
results in the following output:
`error: core.Service is not suitable for converting to "networking.k8s.io/v1" in scheme "pkg/api/legacyscheme/scheme.go:30"`
**What you expected to happen**:

It's fairly obvious _why_ this is happening - the Service doesn't have an appropriate apiVersion.

I'm curious how one might go about skipping mismatches in a master manifest (think something that kustomize might create).

**Environment**:
- `Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.14"`
- `v1.26.2` is the pluggin version I'm using
- Cloud provider or hardware configuration: (n/a - it's local)
- OS (e.g: `cat /etc/os-release`): `PRETTY_NAME="Ubuntu 22.04.2 LTS"`

### PR DESCRIPTION
**Pull request by gxwilkerson33**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This PR solves issue of kubectl-convert failing when resources cannot be converted to desired output version. The solution is to warn the user and continue converting the rest of the file.
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1383
#### Does this PR introduce a user-facing change? No

```release-note
Fixed issue where kubectl-convert would fail when encountering resources that could not be converted to the specified api version. New behavior is to warn the user of the failed conversions and continue to convert the remaining resources. 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by gxwilkerson33** [comment]
cc @eddiezane

File: pkg/kubectl/cmd/convert/convert.go [hunk 1]
```diff
@@ -263,7 +264,10 @@ func asVersionedObjects(infos []*resource.Info, specifiedOutputVersion schema.Gr
 
 		converted, err := tryConvert(scheme.Scheme, info.Object, targetVersions...)
 		if err != nil {
-			return nil, err
+			//Dont fail on error converting objects.
+			//Simply warn the user with the error returned from api-machinery and continue with the rest of the file
+			fmt.Fprintln(os.Stderr, err.Error())
```
Comment (ardaguclu) [inline pkg/kubectl/cmd/convert/convert.go:269]: I think, type of error matters. Perhaps we can only continue if the error is notfound error?

Comment (gxwilkerson33) [reply pkg/kubectl/cmd/convert/convert.go:269]: @ardaguclu yep. i think you are right. i will update this PR with that. Thanks

File: test/fixtures/pkg/kubectl/cmd/convert/serviceandingress.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: ClusterIPExample
+  type: ClusterIP
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: testIngress
```
Comment (eddiezane) [inline test/fixtures/pkg/kubectl/cmd/convert/serviceandingress.yml:18]: nit: same as above. No capitals.

**Review by eddiezane — COMMENTED** [review]
This is looking good but I'm not sure what the expected behavior should be.

Right now the output for multiple resources is wrapped in a v1 List.

```
apiVersion: v1
kind: List
metadata: {}
items:
- apiVersion: v1
  kind: Service
...
```

This is valid but IMO unexpected. I think users would want to pipe the output to a file with only the conversions done with a clean diff. i.e. `kubectl convert -f my-old-network.yaml > my-new-network.yaml`.

@ardaguclu @soltysh what do you think?

File: pkg/kubectl/cmd/convert/convert.go [hunk 3]
```diff
@@ -263,6 +263,14 @@ func asVersionedObjects(infos []*resource.Info, specifiedOutputVersion schema.Gr
 
 		converted, err := tryConvert(scheme.Scheme, info.Object, targetVersions...)
 		if err != nil {
+			//Dont fail on not registered error converting objects.
```
Comment (eddiezane) [inline pkg/kubectl/cmd/convert/convert.go:266]: nit: these comments should have a space. i.e

```suggestion
			// Dont fail on not registered error converting objects.
```

**Comment by ardaguclu** [comment]
I think we can discard the ones not supported(we will already show a warning to user about they will be skipped), List will only contain the ones supported.

**Review by eddiezane — APPROVED** [review]
> I think we can discard the ones not supported(we will already show a warning to user about they will be skipped), List will only contain the ones supported.

It looks like the current behavior is for multiple entries to be wrapped in a v1 List regardless. That's not what I would expect when using this tool but I am ok with it.

@gxwilkerson33 thanks for this PR! It will merge in a week or so after [thaw](https://www.kubernetes.dev/resources/release/#timeline).

/approve
/lgtm

**Pull request merged**